### ISSUE TITLE
False Positives for Smile Detection

### ISSUE THREAD

**Issue by kaisark** [Vishu-Chaudhary/Blink_and_Smile_Detection#1]
Cool Project. I ran into some false positives for smile detection. It seemed like the code was picking up my neutral mouth position (closed) as a smile... (MOUTH_AR_THRESH = 0.3)

![computervision gif-downsized_large](https://user-images.githubusercontent.com/3580312/44999291-240d5000-af8a-11e8-957d-5b18823f513d.gif)

Smile detection with Haar Cascades appears more accurate (faster too?), though I did notice more false positives (not in gif) when moving closer to the camera. Maybe, a deep learning approach for emotion recognition might be best (e.g. Affectiva).

https://github.com/LiuXiaolong19920720/smile-detection-Python

![opencv gif-downsized_large](https://user-images.githubusercontent.com/3580312/45000728-7142ef00-af95-11e8-8845-22c964830aae.gif)

**Comment by vsanse** [comment]
Yes, this one is not accurate. The idea behind project was to test if ratio of points only on lips can be be used for emotions detection. But turns out they alone are not sufficient and we need more concrete formula for getting accurate results. You may want to use other points on face to design it more accurately. For example it detects open mouth as smiling here you can try adding ratio of points on Jawline as the difference in the displacement when mouth is opened vs when person is smiling will be different (just guessing, give it a try).
